tests/geometry/functional/test_so3.py

```diff
@@ -84,3 +84,15 @@ def test_compose(batch_size: int, dtype: torch.dtype):
 
     # check analytic backward for the operator
     check_lie_group_function(so3, "compose", TEST_EPS, group0, group1)
+
```

Comment:
Once again duplicated code. You can reuse the main skeleton of all of these by adding some "preprocess" functions and select the right one for each operation you are checking.

Reply:
Create an issue #402.